packages/server/src/order/order.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { CartItem } from './cart-item.interface';
+import { CheckoutDetails } from './checkout-details.interface';
+import { OrderDto } from './dto/order.dto';
+import { Order } from './order.interface';
+
+@Injectable()
+export class OrderService {
+  constructor(
+    @InjectModel('Order') private readonly orderModel: Model<Order>,
+  ) {}
+
+  findAll() {
+    return this.getAll();
+  }
+
+  find(id: string) {
+    return this.get(id);
+  }
+
+  add(orderDto: OrderDto) {
+    const order: Order = {
```

Comment:
improve naming here, generic names are not preferred. instead of `add` use `addProduct` here.